Objects/unicodeobject.c

```diff
@@ -114,7 +114,7 @@ NOTE: In the interpreter's initialization phase, some globals are currently
 
 static inline char* _PyUnicode_UTF8(PyObject *op)
 {
-    return (_PyCompactUnicodeObject_CAST(op)->utf8);
+    return FT_ATOMIC_LOAD_PTR_ACQUIRE(_PyCompactUnicodeObject_CAST(op)->utf8);
```

Comment:
I don't love the atomic load here. It's not necessary for most of the call sites and I think it makes it harder to see the synchronization, but we can deal with that in follow up PRs.